In [1]:
import pandas as pd

In [2]:
dataset=pd.read_csv("Placement_preprocessed.csv")

In [3]:
dataset.isnull().sum()

gender            0
ssc_b             0
hsc_b             0
hsc_s             0
degree_t          0
workex            0
specialisation    0
status            0
sl_no             0
ssc_p             0
hsc_p             0
degree_p          0
etest_p           0
mba_p             0
salary            0
dtype: int64

In [4]:
dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 215 entries, 0 to 214
Data columns (total 15 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   gender          215 non-null    object 
 1   ssc_b           215 non-null    object 
 2   hsc_b           215 non-null    object 
 3   hsc_s           215 non-null    object 
 4   degree_t        215 non-null    object 
 5   workex          215 non-null    object 
 6   specialisation  215 non-null    object 
 7   status          215 non-null    object 
 8   sl_no           215 non-null    float64
 9   ssc_p           215 non-null    float64
 10  hsc_p           215 non-null    float64
 11  degree_p        215 non-null    float64
 12  etest_p         215 non-null    float64
 13  mba_p           215 non-null    float64
 14  salary          215 non-null    float64
dtypes: float64(7), object(8)
memory usage: 25.3+ KB


In [5]:
dataset=dataset.drop('sl_no',axis=1)

In [6]:
dataset.cov(numeric_only=True)

ssc_p          hsc_p       degree_p        etest_p  \
ssc_p        117.228377      60.348373      42.897137      37.659225   
hsc_p         60.348373     118.755706      34.819820      35.461678   
degree_p      42.897137      34.819820      54.151103      21.929469   
etest_p       37.659225      35.461678      21.929469     176.251018   
mba_p         24.535952      22.555846      17.272020      16.886973   
salary    901754.893936  763359.777657  465131.504238  384234.419257   

                  mba_p        salary  
ssc_p         24.535952  9.017549e+05  
hsc_p         22.555846  7.633598e+05  
degree_p      17.272020  4.651315e+05  
etest_p       16.886973  3.842344e+05  
mba_p         34.028376  1.262455e+05  
salary    126245.485547  2.395714e+10

1. The Covariance between the columns degree_p and etest_p is 21.929469 which is a positive value. In this case degree_p value is directly proportional to etest_p i.e. if one value increases the other will also increase.
2. The Covariance between the columns etest_p and mba_p is 16.886973 which is a positive value. In this case etest_p value is directly proportional to mba_p i.e. if one value increases the other will also increase.

In [7]:
dataset.corr(numeric_only=True)


ssc_p     hsc_p  degree_p   etest_p     mba_p    salary
ssc_p     1.000000  0.511472  0.538404  0.261993  0.388478  0.538090
hsc_p     0.511472  1.000000  0.434206  0.245113  0.354823  0.452569
degree_p  0.538404  0.434206  1.000000  0.224470  0.402364  0.408371
etest_p   0.261993  0.245113  0.224470  1.000000  0.218055  0.186988
mba_p     0.388478  0.354823  0.402364  0.218055  1.000000  0.139823
salary    0.538090  0.452569  0.408371  0.186988  0.139823  1.000000

1. The correlation between the columns mba_p and salary is 0.139823 which is a positive value. Since the value is only 14% therefore both the columns are weakly correlated.